vispy/scene/widgets/grid.py

```diff
@@ -163,6 +163,12 @@ def _update_child_widgets(self):
         cols = np.linspace(rect.left, rect.right, n_cols+1)
         colstart = cols[:-1] + s2
         colend = cols[1:] - s2
+        
+        # snap to pixel boundaries to avoid drawing artifacts
+        colstart = np.round(colstart)
+        colend = np.round(colend)
+        rowstart = np.round(rowstart)
+        rowend = np.round(rowend)
```

Comment:
Do these need to be `int` or `float`? `round` returns a `float`, so just wanted to check.